styles/util/_layout.scss

```diff
@@ -1,20 +1,159 @@
 @use '../mixins/layout';
 
+/* Spacing classes */
+
+// These classes are applied to a parent element to establish horizontal
+// or vertical spacing between immediate-child elements.
+
+// Apply default horizontal spacing
+.hyp-u-horizontal-spacing {
+  @include layout.horizontal-spacing;
+}
+
+// Apply default vertical spacing
+.hyp-u-vertical-spacing {
+  @include layout.vertical-spacing;
+}
+
+// Generate horizontal-spacing utility classes, 0-9
+@for $space-unit from 0 through 9 {
+  .hyp-u-horizontal-spacing--#{$space-unit} {
+    @include layout.horizontal-spacing($size: $space-unit);
+  }
+}
+
+// Generate vertical-spacing utility classes, 0-9
+@for $space-unit from 0 through 9 {
+  .hyp-u-vertical-spacing--#{$space-unit} {
+    @include layout.vertical-spacing($size: $space-unit);
+  }
+}
+
+/* Margin and padding classes */
+
+// These classes are applied to individual elements to override margin or padding
+
+// Default-sized margin classes
+.hyp-u-margin {
+  @include layout.margin;
+}
+
+.hyp-u-margin--left {
+  @include layout.margin($side: 'left');
+}
+
+.hyp-u-margin--right {
+  @include layout.margin($side: 'right');
+}
+
+.hyp-u-margin--top {
+  @include layout.margin($side: 'top');
+}
+
+.hyp-u-margin--bottom {
+  @include layout.margin($side: 'bottom');
+}
+
+// Generate margin utility classes for spacing units 0-9
+@for $space-unit from 0 through 9 {
```

Comment:
The combinatorial size here (10 spacing levels * 5 margin sides) is a bit unfortunate, especially since I'd guess that many of the combinations will not be used. What is the thinking behind utility classes for this vs `margin: var.size($size)` (or `margin-$side: $size`) in component-specific classes?

Reply:
Yes. Hah. Not a yes or no question, I realize.

Your question gets right to the nugget of what I'm working through solving right now. In real-world use, what's the right balance between ergonomics, sensible performance/efficiency (i.e. the combinatorial size issue here), consistency (not having every component be authored differently based on developer whim) and elegance? It remains to be seen. Will `.hyp-u-padding--right--4` be a life-saver, or a ridiculous extra detail glutting our CSS builds?